### DISCUSSION TITLE
Working Examples of Config Recognizing Custom Compiler

### DISCUSSION THREAD

**Discussion by ewhac** [rizsotto/Bear#598] [Q&A]
By default, Bear recognizes when the system's default compilers are invoked.  It would be nice if Bear also recognized other compilers, such as deb-packaged cross-compilers, or compilers packaged as part of an "IDE" (usually an Eclipse knock-off) furnished by a microcontroller/SoC vendor.

Bear allegedly supports recognizing such compilers, but for the life of me I can't figure out how to set up a Bear config to successfully do this.  Google has been no help locating clues.  Could someone please post some working examples of such?

**Comment by rizsotto** [comment]
Hey @ewhac , are you still struggle with this issue?

Two things that needs to be aligned: 

1. The intercept part, when the compiler execution is captured. This can be either compiler wrappers or by the preloaded library. Depends on which one you use, there could be different solutions. (I would recommend to read the [troubleshooting wiki page](https://github.com/rizsotto/Bear/wiki/Troubleshooting) for more context.) It's a safe bet if you create a new link to the `wrapper` program in the `wrapper.d` directory with the compiler name you expect to capture. (There are already soft links in that directory, so just follow the pattern.)
2. The recognize part, when the intercepted program executions will be "recognised" as compiler calls. Here you could create a [config file](https://github.com/rizsotto/Bear/blob/master/source/citnames/man/bear-citnames.1.md#config-file) and in the `compilers_to_recognize` section you list your compilers.

And you can observe how Bear works by passing `-vvvv` as argument, so it will print verbose logs that you can read and observe both stages of the compilation database generation.
